src/shared/invite-response/invite-response.tsx

```diff
@@ -80,15 +116,46 @@ const InviteResponse: FC<InviteResponse> = ({
 		[invite]
 	);
 
-	const participants = useMemo(
+	const [maxReqParticipantsToShow, setMaxReqParticipantsToShow] = useState(4);
+	const requiredParticipants = useMemo(
 		() =>
-			invite[0]?.comp[0].at.map((user: any, index: number) =>
-				index === (invite[0]?.comp[0].at.length ?? 1) - 1 // the nullish coalescing is
-					? `${user.a || user.d}`
-					: `${user.a || user.d},`
-			),
+			invite[0]?.comp[0].at
+				.filter((user: any) => user.role === 'REQ')
+				.map((user: any) => ({
+					name: user.d,
+					email: user.a
+				})),
 		[invite]
 	);
+
+	const [maxOptParticipantsToShow, setMaxOptParticipantsToShow] = useState(5);
+	const optionalParticipants = useMemo(
+		() =>
+			invite[0]?.comp[0].at
+				.filter((user: any) => user.role === 'OPT')
+				.map((user: any) => ({
+					name: user.d || user.a,
+					email: user.a
+				})),
+		[invite]
+	);
+
```

Comment:
user type is simple it can be properly typed 
type Participant ={
a: string;
d: string;
ptst: "NE|AC|DE" (so on)
role:"OPT|REQ" (so on);
rsvp: boolean;
url: string;
}